### PR TITLE
Next.js boilerplate for App Hosting

### PR DESCRIPTION
**Pull request by leoortizz**
Next.js boilerplate for Firebase App Hosting including features like:
 - SSG
 - SSR
 - ISR
 - Streaming

### CODE REVIEW

**Review by jhuleatt — CHANGES_REQUESTED** [review]
Hi @leoortizz thanks for creating the PR! Please move from `starters/basic/nextjs` to `starters/nextjs/basic`. We'd like the framework name to be the top level under "starters"

**Comment by leoortizz** [comment]
> Hi @leoortizz thanks for creating the PR! Please move from `starters/basic/nextjs` to `starters/nextjs/basic`. We'd like the framework name to be the top level under "starters"

oops, my mistake. Just changed it

**Review by jhuleatt — APPROVED** [review]
Sample looks good to me. Seems like CI is failing - is that an easy fix?

**Comment by Yuangwang** [comment]
Could you add a brief description of what this is for?

File: starters/nextjs/basic/src/app/ssg/page.tsx [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import { getDateString, getRandomUUID } from "../utils";
+
+export default function Page() {
+  return (
+    <main className="content">
+      <h1 className="heading">Sample title for SSG</h1>
```
Comment (Yuangwang) [inline starters/nextjs/basic/src/app/ssg/page.tsx:6]: Should this be updated to something else?

**Review by Yuangwang — APPROVED** [review]
Small comment otherwise lgtm once the CI is fixed and the comment is addressed

File: starters/nextjs/basic/src/app/ssg/page.tsx (hunk 1 above)
Comment (leoortizz) [reply starters/nextjs/basic/src/app/ssg/page.tsx]: good catch, thanks, kept it as just SSG

**Comment by leoortizz** [comment]
@Yuangwang I fixed one of the build action failures by running `npm install` and updating the lock file in e821affa73cf8485482cafd324e76831abe56104.

[Now what's failing is `@apphosting/adapter-angular` build script](https://github.com/FirebaseExtended/firebase-framework-tools/actions/runs/8972233763/job/24639762163?pr=173):
```bash
error TS6053: File '/home/runner/work/firebase-framework-tools/firebase-framework-tools/packages/@apphosting/adapter-angular/src/simple-server/rollup.config-1715012009185.cjs' not found.
  The file is in the program because:
    Matched by include pattern 'src/simple-server/*' in '/home/runner/work/firebase-framework-tools/firebase-framework-tools/packages/@apphosting/adapter-angular/tsconfig.json'
npm ERR! Lifecycle script `build` failed with error: 
npm ERR! Error: command failed 
npm ERR!   in workspace: @apphosting/adapter-angular@17.2.3 
npm ERR!   at location: /home/runner/work/firebase-framework-tools/firebase-framework-tools/packages/@apphosting/adapter-angular 
```

both errors I had doesn't seem related to the changes in this PR though, any idea of what's wrong?

FWIW `lerna build` is working locally

**Comment by Yuangwang** [comment]
Yeah that does seem unrelated, looks like something is broken in the angular adapter. I'm not sure why this wasn't caught earlier and is only showing up now.

I'll need to take a closer look later but for now feel free to ignore the error.

**Pull request merged**